includes/classes/Indexable/Term/SyncManager.php

```diff
@@ -40,7 +40,7 @@ public function setup() {
 		add_action( 'added_term_meta', [ $this, 'action_queue_meta_sync' ], 10, 2 );
 		add_action( 'deleted_term_meta', [ $this, 'action_queue_meta_sync' ], 10, 2 );
 		add_action( 'updated_term_meta', [ $this, 'action_queue_meta_sync' ], 10, 2 );
-		add_action( 'delete_term', [ $this, 'action_sync_on_delete' ] );
+		add_action( 'pre_delete_term', [ $this, 'action_sync_on_delete' ] );
```

Comment:
Is `pre_delete_term` too early of a hook? 

Would it be better to use a later hook such as `deleted_term_taxonomy` hook which fires right after the deletion occurs to ensure that the term is actually deleted on the DB level right before ES deletes on its side? https://github.com/WordPress/WordPress/blob/243e75eb416657464870ea9092c8750b7221b8fc/wp-includes/taxonomy.php#L2039-L2046

Reply:
That was my initial thought too, but that hook happens after the child terms were reassigned to another parent. So say you have this scenario if using `deleted_term_taxonomy`:
1) A <- B <- C
2) B is being deleted
3) `pre_delete_term` is executed (we do nothing)
4) C is assigned as a child to A
5) `deleted_term_taxonomy` is executed. We look up B's children and find nothing.
6) Delete B
7) END (C was not scheduled for reindexing)


vs  `pre_delete_term` 
1) A <- B <- C
2) B is being deleted
3) `pre_delete_term` is executed.  We look up B's children and find C.
4) C is assigned as a child to A
5) `deleted_term_taxonomy` is executed. We do nothing
6) Delete B
7) END
8) On shutdown we will reindex C.


I guess another solution is to split the two things into two hooks do the "lookup term being deleted children and queue them up"  on `pre_delete_term` and leave "delete the term" on `delete_term`. What do you think @rebeccahum ?

Reply:
Oh, good point with the nesting hierarchies!

Breaking it up into two separate hooks seems like a good idea to me.

Reply:
Done.